air-quality-backend/etl/src/forecast/forecast_texture_storer.py

```diff
@@ -0,0 +1,163 @@
+from typing import List, Dict, Tuple
+import os
+import pandas as pd
+import xarray as xr
+import numpy as np
+from PIL import Image
+
+
+def _chunk_data_array(
+    rgb_data_array: np.ndarray,
+    num_time_steps: int,
+    num_lon: int,
+    time_vector: xr.DataArray,
+) -> Tuple[List[np.ndarray], Dict[int, Dict[str, str]]]:
+    """
+    Split the rgb_data_array into smaller parts along the num_lon * num_time dimension
+    for smaller file sizes and faster loading times.
+
+    :param rgb_data_array: array with shape (num_lat, num_lon * num_time, channels)
+    :param num_time_steps: Number of time steps to use for each chunk
+    :param num_lon: Number of longitude points
+    :param time_vector: DataArray containing time information for each time step
+    :return: A list of chunked arrays and a dictionary with start and end time
+    stamps for each chunk
+    """
+    num_lat, total_lon_time, channels = rgb_data_array.shape
+    num_total_time_steps = total_lon_time // num_lon
+
+    chunks = []
+    time_steps_dict = {}
+
+    if num_time_steps <= 0 or num_time_steps > num_total_time_steps:
+        num_time_steps = num_total_time_steps
+
+    for i in range(0, num_total_time_steps, num_time_steps):
+        start_time_step = i
+        end_time_step = min(i + num_time_steps, num_total_time_steps)
+
+        start_index = start_time_step * num_lon
+        end_index = end_time_step * num_lon
+
+        chunk = rgb_data_array[:, start_index:end_index, :]
+        chunks.append(chunk)
+
+        time_start = pd.to_datetime(
+            time_vector.data[start_time_step:end_time_step].min(), unit="s"
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        time_end = pd.to_datetime(
+            time_vector.data[start_time_step:end_time_step].max(), unit="s"
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        time_steps_dict[len(chunks) - 1] = {
+            "time_start": time_start,
+            "time_end": time_end,
+        }
+
+    return chunks, time_steps_dict
+
+
+def _create_output_directory(forecast_date: str) -> str:
+    """
+    Create output directory for data textures.
+
+    :param forecast_date: The forecast date timestamp.
+
+    :return: The output directory path and forecast date.
+    """
+
+    output_directory = f"/app/data_textures/{forecast_date}"
```

Comment:
I might be missing something but should this 
f"/app/data_textures/{forecast_date}" 
be 
f"/{os.cwd}/data_textures/{forecast_date}"

?

Reply:
I think this is correct as it is. The idea is that `/app/data_textures/` is the default directory in the shared Docker volume. The f"{os.getcwd()}/data_textures/{forecast_date}" is the fallback to also run the etl outside of Docker and then save the textures to the local directory. This is quite helpful for me for debugging